debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorConfig.java

```diff
@@ -904,6 +915,12 @@ protected String slotName() {
     }
 
     protected boolean dropSlotOnStop() {
+        if (getConfig().hasKey(DROP_SLOT_ON_STOP.name())) {
+            return getConfig().getBoolean(DROP_SLOT_ON_STOP);
+        }
+        if (getConfig().hasKey(DROP_SLOT_ON_STOP_OBSOLETE.name())) {
+            return getConfig().hasKey(DROP_SLOT_ON_STOP_OBSOLETE.name());
+        }
         return getConfig().getBoolean(DROP_SLOT_ON_STOP);
```

Comment:
Hum, this seems redundant with the first one above?

Reply:
Ah, is it to apply the default? If so, a comment would be good, it looks a bit confusing.

Reply:
Comment added